### ISSUE TITLE
Fix issues with file upload feature

### ISSUE THREAD

**Issue by dryhten** [elifesciences/elife-xpub#549]
The upload progress is not completely done yet. 

The list of tasks that still need to be done:

- [x] #544 Delay connection to websocket until token is present in local storage
- [x] #548 Don't show manuscript upload progress for now
- [x] #546 Prevent the server from sending all push back messages at once
- [x] #551 Show upload progress on dropzone molecule again
- [x] #538 Upload Manuscript flaky
- [x] #546 Prevent server from sending all pushback messages at once
- [ ] #493 Fix resolvers 'extracts title from PDF' test
- [ ] #492 Subscribe error handling
- [ ] #652 Upload Progress for 90M file pauses (Bug)
- [ ] #653 Upload Progress does not resume after navigating away (Bug)
- [ ] #639 Refactor component to be stateful
- [ ] #625 Differentiate between upload errors
- [ ] #646 Fix border color when reuploading manuscript (Bug)
- [ ] #641 Perform proper check for file size limit for manuscripts on the backend (Bug)
- [ ] #839 Sort ScienceBeam extraction once and for all

**Comment by hdrury1** [comment]
@diversemix I would like to sit down and figure out which of these are MVP tasks.

**Comment by jsms90** [comment]
I would suggest that we probably shouldn't go to production with some of these bugs e.g. #646 means that a user sees errors when there are none & #653 could make it very unclear which file has been uploaded / whether a file has been uploaded (see @diversemix's comment on the issue).

If an author isn't convinced that the right manuscript has been uploaded, they may not trust the app enough to submit through it. So this feature, more than most, seems like a key part of the app to ensure is as bug-free as possible?

It might be worth going over the user journeys for this particular feature from scratch, considering each possible action and what a user should see when they move from one situation to another? I would be more than happy to sit down with you & @chugginselifesciences and go through the flow some time, if that would help?

**Comment by diversemix** [comment]
I should have updated this... but @hdrury1 and I decided that we'd look at this with the multiple file upload feature after our current set of work... i.e. Production Readiness and integration with eLife

**Comment by jsms90** [comment]
Yeah, it's in the roadmap next to the multiple file upload. To be built after the current work & after the People Picker. :+1: I just meant for the actual product/design phase of the File Upload (before it's then ready to make it into 'Now', in terms of build).

It's totally up to you guys & how you want to work, but it can be useful to have a dev in the room for certain portions of the design phase. Certainly not for the whole thing, but at certain key points. Especially since this feature is already half-built. It may be helpful for one of us to catch @chugginselifesciences up with what we've already got, so the next design-build of this feature is as smooth as poss?

It seems like many of the bugs we've introduced & are raising as github issues, are more about user journeys that we, as a team, hadn't considered before building. So perhaps tighter collaboration here could be beneficial?

I may be jumping the gun, since we have got the People Picker to build first? But it seemed like the design phase for that was largely considered done (aside from the obvious additional questions that come up during build) & so perhaps File Upload was next on @chugginselifesciences' plate in terms of xpub?
